### ISSUE TITLE
Game hears "fists" when you don't say it.

### ISSUE THREAD

**Issue by CarlKenner** [KozGit/DOOM-3-BFG-VR#98]
> also i dunno if there's something up with the control scheme, but it seems that when in the middle of combat my character seems to sometimes change to "fists" during a fight. i disabled auto weapon switch and  made sure "fists" didnt have a key binding to weapon swap to so i dont know how it keeps switching to them

**Comment by KozGit** [comment]
Carl, do I remember reading you did something to address this?

If not, should we maybe just change the recognized phrases for fist(s) to something else? ( maybe 'weapon fists'? 'turkey puncher'? :) )

**Comment by CarlKenner** [comment]
No, I haven't done anything to address this. I only posted some things a user could do if they were having issues with this.

Maybe we should increase the accuracy and volume required for phrases that get misrecognised easily?

Surprisingly, the other phrase it tends to hear is "heart of hell".

**Comment by KozGit** [comment]
I think we should probably do something - if you think putting in a volume/accuracy check would help I'm all for it.  I don't think I've run into an errant 'heart of hell' yet , but I haven't been testing the expansion recently either.
 
( I was just playing, and the kind of hissing door opening sound playing through the rift headphones was just picked up by the mic and triggered the fists - must be too quiet in the house right now. )

**Comment by CarlKenner** [comment]
The other option is to add words that sound like "fists" but aren't, to the word list for talk. For example, "hiss". That would probably reduce misrecognition.

Saying "heart of hell" brings up the soul cube in the main game. They're the only weapons I didn't bother creating two unique codes for. Maybe I should.

**Comment by CarlKenner** [comment]
OK, I separated out The Artifact and Soul Cube into two separate impulses: _soulcube and _artifact. And yes, I'm giving new impulses sensible names now. That should prevent changing to the soul cube in the main game when you didn't say anything. In RoE you might still change to The Artifact when you didn't say anything.
I added "hiss", but it's never recognised, so I don't know if that helped.

When i say "fist", it's often misrecognised as "pistol" with high confidence but low volume (0 or 1 out of 100). But when I actually say "pistol", it's recognised as "pistol" with high confidence and correct volume (usually > 2 unless I whisper). I could add a hack so that "pistol" requires volume >= 2, but I don't know if misrecognising "fist" as nothing is better than misrecognising "fist" as "pistol". Or I could add a really nasty hack that recognises "pistol" with low volume as "fist". Try using vr_voiceRepeat 1 and see if you get the same behaviour, or if it's unique to me.

When "fist" is correctly recognised as "fist", it's also low volume. I guess short hissing sounds don't trigger the VU meter, but the "ol" sound in "pistol" does. So I can't require volume for all commands or it will never hear "fist".

**Comment by jdawgzim** [comment]
I play HTC Vive with external speakers on and it switches to fists and other things often.  Also I often have other people in the room I'm playing that are talking.

SOLUTION: I downloaded OpenVR - Advanced Settings and was able to create a push-to-talk button for Doom 3 VR voice commands. Problem with this is it works for all games so I have to turn it on then off every time I play Doom 3 VR.
https://github.com/matzman666/OpenVR-AdvancedSettings

REQUEST: Please add Push-to-talk button for Voice Commands in Doom 3 VR
